core/src/test/scala/integration/kafka/api/SaslEndToEndAuthorizationTest.scala

```diff
@@ -79,7 +79,7 @@ abstract class SaslEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
       consumeRecords(consumer2)
       fail("Expected exception as consumer2 has no access to group")
     } catch {
-      case _: GroupAuthorizationException => //expected
+      case _: TopicAuthorizationException => //expected
```

Comment:
We should update the failure message above. By the way, I'm curious why we hit the topic exception first.

Reply:
Previously we used to get UNKNOWN_TOPIC_OR_PARTITION for metadata fetch request and GroupAuthorizationException during Join group request.  

Now, we are getting TopicAuthorizationException during metadata fetch request and fail immediately.